pkg/server/server.go

```diff
@@ -92,15 +92,8 @@ func (s *server) RunUntil(stopCh <-chan struct{}) error {
 	defer cancel()
 
 	// Start informers
-	go s.nodes.Run(stopCh)
 	go s.pods.Run(stopCh)
-
-	// Ensure cache is up to date
-	ok := cache.WaitForCacheSync(stopCh, s.nodes.HasSynced)
```

Comment:
Why remove wait for cache? I expect that Apiserver starts the informer, however I didn't see this mentioned by you.

Reply:
let me update it

Reply:
I try to recover the following codes
````
ok := cache.WaitForCacheSync(stopCh, s.nodes.HasSynced)
if !ok {
return nil
}
````
But metrics-server doesn't start properly.
The reason is that the apiserver is started after `cache.WaitForCacheSync`, so it is always blocking at `WaitForCacheSync`

Reply:
Waiting for node cache sync is important, we should not remove it as it will increase initialization time. If caches are not ready, the first scrape round will not scrape any nodes, so instead MS being ready after 15s, it will need 30s (default 15s metric-resolution).

Please consider some rewrite to avoid getting blocked. One idea I have is to pass different informer argument to Apiserver. Maybe `podInformer` one or `nil`.

Reply:
yeah,  Let me try to solve it.

Reply:
I think this solution is awesome. Thanks for providing a perfect solution.